lib/scheduler/thread_pool.rb

```diff
@@ -8,15 +8,19 @@ module Scheduler
   # Usage:
   #  pool = ThreadPool.new(min_threads: 0, max_threads: 4, idle_time: 0.1)
   #  pool.post { do_something }
+  #  pool.stats (returns thread count, busy thread count, etc.)
   #
-  #  (optional)
-  #  pool.shutdown
+  #  pool.shutdown (do not accept new tasks)
+  #  pool.wait_for_termination(timeout: 1) (optional timeout)
 
   class ThreadPool
     class ShutdownError < StandardError
     end
 
-    def initialize(min_threads:, max_threads:, idle_time:)
+    def initialize(min_threads:, max_threads:, idle_time: nil)
+      # this is there just for cases where people want to use this as
+      # a fixed size thread pool
+      idle_time ||= 30
```

Comment:
Is there something significant about the number `30` as I'm wondering why setting it to 30 would mean it become a fixed size thread pool

Reply:
will add to the comment, basically in a "fixed" mode stuff still idles per the current design, so this is a default.